src/sidebar/components/filter-status.js

```diff
@@ -0,0 +1,156 @@
+import { createElement } from 'preact';
+
+import { countVisible } from '../util/thread';
+
+import Button from './button';
+
+import useRootThread from './hooks/use-root-thread';
+import useStore from '../store/use-store';
+
+/**
+ * Render information about the currently-applied filters and allow the
+ * user to clear filters or toggle user-focus mode on and off
+ *
+ * When any filtering is applied, we are in one of 3 mutually-exclusive filtering
+ * "modes". In any mode, we display descriptive text about what the user
+ * is looking at, and present one button to take action on the current filtering
+ * state:
+ * - selection: One or more annotations have been selected in the document by
+ *   the user, or we're looking at a direct-linked annotation. Button clears
+ *   the selection. This mode supersedes all others (i.e. other filters ignored).
+ * - query: There is a user-entered filter query. It can "stack" with user-focus.
+ *    Button clears the selection (query, etc.), but will leave user-focus mode
+ *    intact.
+ * - focus: Neither of the above two modes apply and user-focus mode is
+ *   configured (it may or may not be active). Button toggles activation of
+ *   user-focus mode.
```

Comment:
It sounds from reading this like "focus" is not mutually-exclusive vs. the other modes.

Reply:
Perhaps renaming this state/mode within the component would help with this confusion. Within the component, it _is_ a mutually-exclusive state, but what it means is "user-focus mode _only_". User focus mode combined with a user-entered filter would put the component in "query" mode.

That is, `focus` mode in the component does not correspond with "user-focus mode is configured" . It corresponds with "user-focus mode is configured and nothing else is going on in filter land."

Reply:
Note: In case I forget to mention it, I've rewritten commenting in the entire module to be consistent about the use of the term "mode." _Mode_ now only refers to the three modes defined in the component itself, which are: `selection`, `query`, and `focusOnly`. I renamed the latter mode for clarity, as user-focus might be active in the other two modes, but the `focusOnly` mode indicates that it is the _only_ applied filter.

I've expanded the comments at the top of the component to try to be extra clear here. Thanks for flagging this up. It is/was indeed quite confusing.